File: benchmarks/implementations/swig/swig_benchmarks.py
import swig_benchmarks_wrappers_py

class Benchmarks(swig_benchmarks_wrappers_py.IBasicBenchmarks):
    def __init__(self):
        super(Benchmarks, self).__init__()
        swig_benchmarks_wrappers_py.SetGlobalBenchmarks(self)

    def __del__(self):
        swig_benchmarks_wrappers_py.SetGlobalBenchmarks(None)


    def VoidToVoid(self): pass

    def I32ToVoid(self, p): pass
    def VoidToI32(self): return 0

    def StringToVoid(self, s): pass
    def VoidToString3(self): return "abc";
    def VoidToString100(self): return "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890";


    def MeasureNativeVoidToVoid(self, n):
        for i in range(n): self.VoidToVoid()
    def MeasureNativeI32ToVoid(self, n):
        for i in range(n): self.I32ToVoid(0)
    def MeasureNativeVoidToI32(self, n):
        for i in range(n): self.VoidToI32()
    def MeasureNativeString3ToVoid(self, n):
        for i in range(n): self.StringToVoid("abc")
    def MeasureNativeVoidToString3(self, n):
        for i in range(n): self.VoidToString3()
    def MeasureNativeString100ToVoid(self, n):
        for i in range(n): self.StringToVoid("abc")
    def MeasureNativeVoidToString100(self, n):
        for i in range(n): self.VoidToString100()


    def MeasureOutgoingVoidToVoid(self, invokable, n):
        for i in range(n): invokable.VoidToVoid()
    def MeasureOutgoingI32ToVoid(self, invokable, n):
        for i in range(n): invokable.I32ToVoid(0)
    def MeasureOutgoingVoidToI32(self, invokable, n):
        for i in range(n): invokable.VoidToI32()
    def MeasureOutgoingString3ToVoid(self, invokable, n):
        for i in range(n): invokable.StringToVoid("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890")
    def MeasureOutgoingVoidToString3(self, invokable, n):
        for i in range(n): invokable.VoidToString3()
    def MeasureOutgoingString100ToVoid(self, invokable, n):
        for i in range(n): invokable.StringToVoid("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890")
    def MeasureOutgoingVoidToString100(self, invokable, n):
        for i in range(n): invokable.VoidToString100()


def GetBenchmarks():
    return Benchmarks()

File: benchmarks/implementations/swig/SwigBenchmarks.java
public class SwigBenchmarks extends SB.IBasicBenchmarks
{
	private java.util.Random r = new java.util.Random();
	public int dummyInt = 0;

	public SwigBenchmarks()
	{ SB.swig_benchmarks_wrappers_java.SetGlobalBenchmarks(this); }


	public void NativeVoidToVoid() { dummyInt = r.nextInt(); }

	public void NativeI32ToVoid(int p) { dummyInt = r.nextInt() + p; }
	public int NativeVoidToI32() { dummyInt = r.nextInt(); return dummyInt; }

	public void NativeStringToVoid(String s) { dummyInt = r.nextInt() + s.length(); }
	public String NativeVoidToString3() { return "abc"; }
	public String NativeVoidToString100() { return "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"; }


	public void VoidToVoid() { }

	public void I32ToVoid(int p) { }
	public int VoidToI32() { return 0; }

	public void StringToVoid(String s) { }
	public String VoidToString3() { return "abc"; }
	public String VoidToString100() { return "1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"; }


	public void MeasureNativeVoidToVoid(long n)
	{ for (long i = 0; i < n; ++i) NativeVoidToVoid(); }

	public void MeasureNativeI32ToVoid(long n)
	{ for (long i = 0; i < n; ++i) NativeI32ToVoid(42); }

	public void MeasureNativeVoidToI32(long n)
	{ for (long i = 0; i < n; ++i) NativeVoidToI32(); }

	public void MeasureNativeString3ToVoid(long n)
	{ for (long i = 0; i < n; ++i) NativeStringToVoid("abc"); }

	public void MeasureNativeVoidToString3(long n)
	{ for (long i = 0; i < n; ++i) NativeVoidToString3(); }

	public void MeasureNativeString100ToVoid(long n)
	{ for (long i = 0; i < n; ++i) NativeStringToVoid("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"); }

	public void MeasureNativeVoidToString100(long n)
	{ for (long i = 0; i < n; ++i) NativeVoidToString100(); }


	public void MeasureOutgoingVoidToVoid(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.VoidToVoid(); }

	public void MeasureOutgoingI32ToVoid(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.I32ToVoid(0); }

	public void MeasureOutgoingVoidToI32(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.VoidToI32(); }

	public void MeasureOutgoingString3ToVoid(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.StringToVoid("abc"); }

	public void MeasureOutgoingVoidToString3(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.VoidToString3(); }

	public void MeasureOutgoingString100ToVoid(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.StringToVoid("1234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"); }

	public void MeasureOutgoingVoidToString100(SB.IBasicInvokable invokable, long n)
	{ for (long i = 0; i < n; ++i) invokable.VoidToString100(); }
}
